pf/proto/internal/gather/type.go

```diff
@@ -0,0 +1,71 @@
+// Copyright 2016-2024, Pulumi Corporation.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package gather
+
+import (
+	"context"
+
+	"github.com/hashicorp/terraform-plugin-framework/attr"
+	"github.com/hashicorp/terraform-plugin-go/tftypes"
+)
+
+type _type struct{ s tftypes.Type }
```

Comment:
This looks rather suspect why is it needed.

Reply:
Why? I needed a type that would bridge `tftypes.Type` back to the higher level `attr.Type`. `_type` projects a `tftype.Type` into the richer typespace of `attr.Type`.

Reply:
It's needed because `pfutils.Attr` thinks in `attr.Type`, instead of the lower level `tftypes.Type`.

Reply:
That's exactly what's suspect to me, I was surprised to see pfutils used at all here, and looks like that's what's pulling in attr.Type.

Reply:
Refactoring out `pfutils` allowed me to remove this.